src/components/RightSidebar/TableOfContents.tsx

```diff
@@ -19,28 +19,61 @@ const TableOfContents: FunctionalComponent<Props> = ({ headers = [], labels }) =
 		if (!toc.current) return;
 
 		const setCurrent: IntersectionObserverCallback = (entries) => {
+			const intersectingEntries = [];
+
 			for (const entry of entries) {
-				if (entry.isIntersecting) {
-					setCurrentID(entry.target.id);
-					break;
-				}
+				if (entry.isIntersecting) intersectingEntries.push(entry);
+			}
+
+			if (intersectingEntries.length === 1) {
+				setCurrentID(intersectingEntries[0].target.id);
+				scrollIntoView();
+			} else if (intersectingEntries.length > 1) {
+				const sortedEntries = intersectingEntries.sort((a, b) => {
+					return getHeadingIndex(a.target.id) - getHeadingIndex(b.target.id);
+				});
+				setCurrentID(sortedEntries[0].target.id);
+				scrollIntoView();
 			}
 		};
 
 		const observerOptions: IntersectionObserverInit = {
 			// Negative top margin accounts for `scroll-margin`.
 			// Negative bottom margin means heading needs to be towards top of viewport to trigger intersection.
-			rootMargin: '-100px 0% -66%',
-			threshold: 1,
+			rootMargin: '-100px 0px -66% 0px',
+			threshold: [0.75, 1],
```

Comment:
Interesting. What did you find the second `threshold` value helped with?

I’m seeing the highlighted menu item jump back and forth now, presumably as each heading crosses its second threshold:

https://user-images.githubusercontent.com/357379/167094798-be238cab-3889-41a6-8b27-ddb08790798d.mov

Reply:
I'm actually confused right now 😅. For me any of these problems didn't happen! I added the second threshold value to maintain the heading on top selected even if it isn't totally visible but most part of it is.

Maybe removing the `0.75` from the threshold solves this? I think if I make any changes I can't properly test it since I didn't had the same problems. 

Can you guys test changing some values (probably threshold and bottom margin?) so we can iterate over what looks best?

Reply:
Haha, yeah this stuff is tricky! This only shows up I think when two headings are close together so 0.75 for one is passed before 1 for the other. For me just `1` on its own works well?